enumerable.rb

```diff
@@ -0,0 +1,109 @@
+module Enumerable
+  def my_each
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    (0..size - 1).my_each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    # an empty array to store our selected element
+    arr = []
+    (0...size - 1).my_each do |i|
+      arr << self[i] if yield(self[i])
+    end
+    arr
+  end
+
+  def my_all?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 unless yield self[i]
+    end
+    count.zero?
+  end
+
+  def my_any?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 if yield self[i]
+    end
+    count != 0
+  end
+
+  def my_none?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 if yield(self[i])
+    end
+    count.zero?
+  end
```

Comment:
#### my_none? 

- [ ] when no block or argument is given returns true only if none of the collection members is true

     > this [nil, false, true, []].my_none? == [nil, false, true, []].none? => false

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class

     > this [1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => fakse

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex

     > %w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => false

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

     > [3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false

[OPTIONAL]

I think if we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?